library/ssl_srv.c

```diff
@@ -3630,7 +3630,7 @@ static int ssl_decrypt_encrypted_pms( mbedtls_ssl_context *ssl,
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
     mbedtls_pk_context *private_key = mbedtls_ssl_own_key( ssl );
     mbedtls_pk_context *public_key = &mbedtls_ssl_own_cert( ssl )->pk;
-    size_t len = mbedtls_pk_get_len( public_key );
+    uint16_t len = (uint16_t) mbedtls_pk_get_len( public_key );
```

Comment:
Ditto - this instance not worth fixing

Reply:
Just a little below, at line 3657 the `len` variable is used as 16-bit unsigned integer and therefore we would never need the whole `size_t`.
Yet again, type cast is only necessary to suppress a compiler warning.

Reply:
Here, I accept the cast to a smaller type is OK, assuming the length is always represented in less than 16-bits. Even so, it could do with a comment like the earlier PR referring to a spec that validates this assumption.

Reply:
> a comment like the earlier PR

This is interesting. A value could be silently truncated to two bytes, but explicit type cast always raised questions.
According to comments, the returned value is the size of `public_key` in bytes. 16 bits would allow 2^19-1 bits for a key.
Would you mind suggesting a good comment line?

Reply:
Hmm, I'm not the best person to suggest something here. @mpg - can you suggest a comment to use? Presumably we can assume something about the TLS buffer structure format?

Reply:
At this point, it isn't about TLS, it's about what `mbedtls_pk_get_len` can return. The maximum is `MBEDTLS_MPI_MAX_SIZE`, reached for an RSA key, and that value is 1024 by default and we say “Maximum tested: 2048 bytes” in `include/mbedtls/config.h`. That's RSA-16384 which is already slower than most people's patience.

Until post-quantum crypto changes the deal, I'm perfectly happy with assuming that key sizes _in bits_ (let alone in bytes) fit in 16 bits. There's other code in the library that makes this assumption. I'm ok with not checking this assumption in TLS code.